FIXED: Discount fixed amount whole cart applied mutiple time when customer use Check Out with Multiple Addresses

mahesh-rajawat opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->
Fixed Issue with CartFixed and Multishipping.
### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
I have fixed issue #25834 in this PR
### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. #25834 

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Create cart rule, no condition, action: Fixed amount discount for whole cart. Discount amount $10
2. Create account, create 2 addresses: address 1, address 2
3. Add 2 SKUs to cart: SKU1, SKU2
4. Check Out with Multiple Addresses
5. SKU1 ship to address 1, SKU2 ship to address 2
6. Order review page discount applied once. https://prnt.sc/qoju2d 
7. Placed an order discount applied in single order. 

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

mahesh-rajawat committed:
FIXED: Discount fixed amount whole cart applied mutiple time when customer use Check Out with Multiple Addresses

mahesh-rajawat committed:
static variable accessible changes

bubasuma commented:
Hi @maheshWebkul721! Our team started working on this PR in scope of [MC-31800](https://jira.corp.magento.com/browse/MC-31800)

bubasuma committed:
MC-29276: Discount fixed amount whole cart applied mutiple time when customer use Check Out with Multiple Addresses

- Fix "discount fixed amount whole cart" for multi-shipping quote

- Add integration for "Discount fixed amount whole cart" with multiple shipping

bubasuma commented:
I am working on integration test

bubasuma committed:
Merge remote-tracking branch 'origin/2.4-develop' into issue-fixed-25834
